### PR TITLE
Log tapper

### LINKED ISSUE #47: Add log tapping
Client tests currently have no visibility into what is happening on the server and can only do test validation based on the results it gets back. If additional operations needed to happen or a specific action needs to be verified there may be no output which can be checked. Log tapping would involve getting all the server logs which occur during a test and being able to check those logs for validating a test condition.

Logs should be accessible directly from the test runner and be able to only get the logs which occur during the test (relying on timestamps may not be accurate enough).

### PR DESCRIPTION
**Pull request by dmcgowan**
Adds ability for tests to tap into the compose logs during the test run and check for expected output on the tested services.    Adds a new binary to test runner to tapping logs and listens to a unix socket for tap requests.

Closes #47

### CODE REVIEW

**Comment by dmcgowan** [comment]
Probably need docs for this one since its use will not be self explanatory. Here is an example of how it could be used...
```
golem_tapper compose > /tmp/tap.out &
docker pull localregistry:5440/someimage
grep "something" /tmp/tap.out
```

ping @aaronlehmann

File: images/Dockerfile.base [hunk 1]
```diff
@@ -29,6 +29,7 @@ RUN pip install docker-compose==1.6
 
 # Install golem binary
 COPY golem /usr/local/bin/golem_runner
+COPY golem /usr/local/bin/golem_tapper
```
Comment (aaronlehmann) [inline images/Dockerfile.base]: Use a hard link?

File: golem.go [hunk 2]
```diff
@@ -213,3 +230,26 @@ func runnerMain() {
 	logrus.Debugf("Shutting down log router")
 	router.Shutdown()
 }
+
+func tapperMain() {
+	var tapSocket string
+	var stderr bool
+
+	flag.StringVar(&tapSocket, "tap-socket", "/var/run/golem-logs", "Socket to spawn log tapper")
```
Comment (aaronlehmann) [inline golem.go]: Since this is the client, this wording probably isn't quite right.

**Comment by aaronlehmann** [comment]
Looks good in general. I imagine the vast majority of use cases will involve looking for the presence or absence of a specific pattern in the log. It might be useful to allow the client to specify a regexp that can be evaluated on the server side, rather than sending the full log output for the client to evaluate. However, this may be a premature optimization.

File: images/Dockerfile.base (hunk 1 above)
Comment (dmcgowan) [inline images/Dockerfile.base]: I had that thought, let me try it out and make sure it doesn't cause any problems.

**Comment by dmcgowan** [comment]
I do agree about the patterns, I will create an feature issue for it when this gets merged.

**Pull request merged**